### PR TITLE
Update terraform/aws to support terraform version 0.7.x

### PR DESCRIPTION
**Pull request by tamsky**
supersedes #171 

/cc @enxebre

### CODE REVIEW

File: docs/getting-started-guides/aws/public.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ for i in $(ls .terraform/modules/*/Makefile); do i=$(dirname $i); make -C $i; do
 ### Provision the cluster infrastructure
 
 ```
+cd /tmp/kubeform/terraform/aws/public-cloud
```
Comment (enxebre) [inline docs/getting-started-guides/aws/public.md]: This step is not necessary, we can remove this. We need to add the new terraform version requirement in this file.
For squashing I usually use ```git rebase -i``` you can also use ``` git merge --squash``` or something like https://github.com/tj/git-extras/blob/master/Commands.md#git-squash, or just recreate your local branch as you must have done here,  then you need to ```git push --force``` against your branch when you are sure you want to override it

**Comment by tamsky** [comment]
@enxebre I've granted permission for you to commit to this branch in my repo, if you'd prefer to make any further edits.

Also suggest checking out github's "Squash & Merge" option if you'd like to keep the commit history clean.

**Comment by enxebre** [comment]
Hey @tamsky thanks for the suggestion! I'm sorry if my last comment here https://github.com/Capgemini/kubeform/pull/172/commits/bea178174958845c1aba0a274efcd40c13d6a4b8 was confuse, the bit that we should remove is ```terraform apply -target=null_resource.etcd_discovery_url```
We don't need that step.
The file is already in the repo, so ```"${file(var.etcd_discovery_url_file)}"``` won't complain when creating the resource graph.
Then the null resource will populate the file with the right value right before creating the new vm at execution time.
Don't you agree?
After that we can merge!

**Comment by tamsky** [comment]
Just tested now.

> Don't you agree?

I agree!

Squash & merge away.

File: terraform/aws/private-cloud/main.tf [hunk 2]
```diff
@@ -52,20 +52,20 @@ module "elb" {
...
-  vars {
-    size = "${var.masters}"
-  }
+resource "null_resource" "etcd_discovery_url" {
+    provisioner "local-exec" {
+        command = "curl -s https://discovery.etcd.io/new?size=${var.masters} > ${var.etcd_discovery_url_file}"
+    }
+
+    # This will regenerate the discovery URL if the cluster size changes
+    triggers {
```
Comment (tamsky) [inline terraform/aws/private-cloud/main.tf]: do we want to remove this from the `private-cloud` tree as well?

Comment (enxebre) [inline terraform/aws/private-cloud/main.tf]: yes please, lets keep both aligned for consistency.

**Comment by tamsky** [comment]
I have updated the private-cloud tree with the changes I was able to identify.

`terraform plan` now runs cleanly.

Also found [this syntax/typo](https://github.com/Capgemini/kubeform/pull/172/commits/40c0dd8f930478c59f7e0e52cdf269437243f809#diff-cda941a3b19b4ef851fee52fbc6067c5L23) that probably should have been caught before by any test that ran `terraform plan`.

Given that there is no README for this directory, I'm not testing `terraform apply`

@enxebre please test `private-cloud` if your org is still using that directory.

Thanks!

**Comment by tamsky** [comment]
@enxebre also, what do you want to do regarding the outputs that exist in `private-cloud` but are missing from `public-cloud`:

https://github.com/tamsky/kubeform/blob/terraform-0.7.x/terraform/aws/private-cloud/main.tf#L80-L95

**Comment by enxebre** [comment]
@tamsky this is good to go. thanks for contributing!

**Pull request merged**